library/src/main/java/org/xmtp/android/library/Client.kt

```diff
@@ -49,6 +55,8 @@ data class ClientOptions(
     val api: Api = Api(),
     val preCreateIdentityCallback: PreEventCallback? = null,
     val preEnableIdentityCallback: PreEventCallback? = null,
+    val appContext: Context? = null,
+    val enableLibXmtpV3: Boolean = false,
```

Comment:
We need the application to pass a context in from the application layer so that we can create a database. Is the `ClientOptions` the correct place for that?
Some for the flag on if V3 is enabled.

Reply:
I would vote in favor of that, curious what @nakajima and @neekolas think.

By the way, I think we will also need a `preGrantMessagingAccess` callback that gets called before signing the text that comes from v3

Reply:
Can you tell me more about this callback/ how I should use it? I'm assuming I can add it later.

Reply:
Fine to add later. With v3, we're adding a new signature type called 'Grant Messaging Access', it looks like (2) [here](https://github.com/xmtp/libxmtp/blob/main/xmtp_mls/IDENTITY.md#installation-registration), and is what will be passed back from `Client.text_to_sign()`.

`preGrantMessagingAccess` would be the equivalent of `preCreateIdentity`, it gives the app developer a chance to show something in the UI before requesting the signature.

Reply:
Maybe we can give this `enableLibXmtpV3` option a friendlier name, if it's going to be visible to integrators. Like `enableGroups`